air/src/air/mod.rs

```diff
@@ -530,7 +530,7 @@ pub trait Air: Send + Sync {
     /// composition polynomial.
     fn get_deep_composition_coefficients<E, H>(
         &self,
-        public_coin: &mut RandomCoin<Self::BaseField, H>,
+        public_coin: &mut RandomCoinImpl<Self::BaseField, H>,
     ) -> Result<DeepCompositionCoefficients<E>, RandomCoinError>
```

Comment:
`RandomCoin` should be a generic type parameter. This probably also means that `H` would need to be `ElementHasher` rather than just `Hasher`.